x-pack/plugins/observability_solution/synthetics/server/routes/overview_status/overview_status_service.ts

```diff
@@ -0,0 +1,331 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment/moment';
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { SavedObjectsFindResult } from '@kbn/core-saved-objects-api-server';
+import { isEmpty } from 'lodash';
+import { withApmSpan } from '@kbn/apm-data-access-plugin/server/utils/with_apm_span';
+import { asMutableArray } from '../../../common/utils/as_mutable_array';
+import { getMonitorFilters, OverviewStatusQuery } from '../common';
+import {
+  getAllMonitors,
+  processMonitors,
+} from '../../saved_objects/synthetics_monitor/get_all_monitors';
+import { ConfigKey } from '../../../common/constants/monitor_management';
+import { RouteContext } from '../types';
+import {
+  EncryptedSyntheticsMonitorAttributes,
+  OverviewStatusMetaData,
+} from '../../../common/runtime_types';
+import { isStatusEnabled } from '../../../common/runtime_types/monitor_management/alert_config';
+import {
+  FINAL_SUMMARY_FILTER,
+  getRangeFilter,
+  getTimespanFilter,
+} from '../../../common/constants/client_defaults';
+
+type LocationStatus = Array<{
+  status: string;
+  locationId: string;
+  timestamp: string;
+}>;
+
+export const SUMMARIES_PAGE_SIZE = 5000;
+
+export class OverviewStatusService {
+  routeContext: RouteContext<Record<string, any>, OverviewStatusQuery>;
+  filterData: {
+    locationFilter?: string[] | string;
+    filtersStr?: string;
+  } = {};
+  constructor(routeContext: RouteContext) {
+    this.routeContext = routeContext;
+  }
+
+  async getOverviewStatus() {
+    const { request } = this.routeContext;
+    const queryParams = request.query as OverviewStatusQuery;
+
+    this.filterData = await getMonitorFilters({
+      ...queryParams,
+      context: this.routeContext,
+    });
+
+    const [allConfigs, statusResult] = await Promise.all([
+      this.getMonitorConfigs(),
+      this.getQueryResult(),
+    ]);
+
+    const { up, down, pending, upConfigs, downConfigs, pendingConfigs, disabledConfigs } =
+      this.processOverviewStatus(allConfigs, statusResult);
+
+    const {
+      enabledMonitorQueryIds,
+      disabledMonitorQueryIds,
+      allIds,
+      disabledCount,
+      disabledMonitorsCount,
+      projectMonitorsCount,
+    } = processMonitors(allConfigs, this.filterData?.locationFilter);
+
+    return {
+      allIds,
+      allMonitorsCount: allConfigs.length,
+      disabledMonitorsCount,
+      projectMonitorsCount,
+      enabledMonitorQueryIds,
+      disabledMonitorQueryIds,
+      disabledCount,
+      up,
+      down,
+      pending,
+      upConfigs,
+      downConfigs,
+      pendingConfigs,
+      disabledConfigs,
+    };
+  }
+
+  async getQueryResult() {
+    return withApmSpan('monitor_status_data', async () => {
+      const range = {
+        // max monitor schedule period is 4 hours
+        from: moment().subtract(4, 'hours').subtract(20, 'minutes').toISOString(),
```

Comment:
```suggestion
        from: moment().subtract(MAX_MONITOR_PERIOD_SCHEDULE, 'hours').subtract(A_VAR_EXPLAINING_WHY_20_MINS_WORKS_WELL, 'minutes').toISOString(),
```

Would be good to extract these to vars, whether within the class or outside, to help illustrate the reason these particular values make sense. Can also delete the comment above if we do that.